### PR TITLE
Only use Mockito for mocking.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We used various mocking frameworks. We should only use one...

Modifications:

Make usage of mocking framework consistent by only using Mockito.

Result:

Less dependencies and more consistent mocking usage.

### CODE REVIEW

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketClientExtensionHandlerTest.java [hunk 1]
```diff
@@ -27,11 +27,10 @@
 import org.junit.Test;
 
 import static io.netty.handler.codec.http.websocketx.extensions.WebSocketExtensionTestUtil.*;
-import static org.easymock.EasyMock.*;
 import static org.junit.Assert.*;
 
 public class WebSocketClientExtensionHandlerTest {
-
+/*
```
Comment (nmittler) [inline codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketClientExtensionHandlerTest.java]: Disabling the entire test?

File: codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionTest.java [hunk 2]
```diff
@@ -541,6 +542,19 @@ public void listenerThrowShouldNotPreventOtherListenersFromBeingNotified() throw
             reservedStream.open(false);
             verify(clientListener).onStreamActive(streamEq(reservedStream));
             verify(clientListener2).onStreamActive(streamEq(reservedStream));
+            /*
```
Comment (nmittler) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionTest.java]: do you need to do something here? Or delete the code?

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketClientExtensionHandlerTest.java (hunk 1 above)
Comment (normanmaurer) [inline codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketClientExtensionHandlerTest.java]: ups... sorry my fault let me fix this :)

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketClientExtensionHandlerTest.java [hunk 3]
```diff
@@ -27,37 +27,35 @@
 import org.junit.Test;
 
 import static io.netty.handler.codec.http.websocketx.extensions.WebSocketExtensionTestUtil.*;
-import static org.easymock.EasyMock.*;
 import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.atLeastOnce;
```
Comment (netkins) [inline codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketClientExtensionHandlerTest.java]: MAJOR Add some tests to this class. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2187)

File: codec-http2/src/test/java/io/netty/handler/codec/http2/Http2TestUtil.java [hunk 4]
```diff
@@ -464,6 +464,7 @@ public ChannelPromise unvoid() {
                     @Override
                     public void operationComplete(ChannelFuture future) throws Exception {
                         if (!future.isSuccess()) {
+                            future.cause().printStackTrace();
```
Comment (Scottmitch) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/Http2TestUtil.java]: remove?

File: codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorTest.java [hunk 5]
```diff
@@ -929,6 +931,7 @@ private int captureWrites(int streamId) {
     private int captureWrites(Http2Stream stream) {
         ArgumentCaptor<Integer> captor = ArgumentCaptor.forClass(Integer.class);
         verify(writer, atLeastOnce()).write(same(stream), captor.capture());
+        System.err.println(captor.getAllValues().size());
```
Comment (Scottmitch) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorTest.java]: remove?

File: pom.xml [hunk 6]
```diff
@@ -472,33 +472,9 @@
...
-            <groupId>junit</groupId>
-            <artifactId>junit-dep</artifactId>
-          </exclusion>
-        </exclusions>
-      </dependency>
-      <dependency>
         <groupId>org.mockito</groupId>
         <artifactId>mockito-core</artifactId>
-        <version>1.10.19</version>
+        <version>2.7.2</version>
```
Comment (Scottmitch) [inline pom.xml]: woo ... been a while 👍

**Comment by Scottmitch** [comment]
to clarify ... thanks for the cleanup ... a few small comments ... lgtm ... we owe you beers :)

**Comment by trustin** [comment]
Great job, man!

File: codec-http2/src/test/java/io/netty/handler/codec/http2/Http2TestUtil.java (hunk 4 above)
Comment (normanmaurer) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/Http2TestUtil.java]: yes

File: codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorTest.java (hunk 5 above)
Comment (normanmaurer) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorTest.java]: yes

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (a7c0ff665cf47dd4ce98e268fda5fcee335cbbb8) and 4.0 (917952d0a99c121db3a56cc3a1c25eb00f358dec)

**Pull request closed without merging**